### PR TITLE
Fix #172: Support MSSQL

### LINKED ISSUE #172: Support MSSQL
Add respective drivers.

### PR DESCRIPTION
**Pull request by romanstrobl**

### CODE REVIEW

File: docs/db/changelog/changesets/user-data-store/1.3.x/20240514-refactor-uds-1.3.0.xml [hunk 1]
```diff
@@ -33,8 +33,8 @@
             <column name="encryption_mode" type="VARCHAR(255)" defaultValue="NO_ENCRYPTION">
                 <constraints nullable="false"/>
             </column>
-            <column name="timestamp_created" type="TIMESTAMP WITHOUT TIME ZONE" defaultValueComputed="NOW()"/>
-            <column name="timestamp_last_updated" type="TIMESTAMP WITHOUT TIME ZONE"/>
+            <column name="timestamp_created" type="TIMESTAMP WITHOUT TIME ZONE" defaultValueDate="${now}" />
```
Comment (banterCZ) [inline docs/db/changelog/changesets/user-data-store/1.3.x/20240514-refactor-uds-1.3.0.xml:36]: @zcgandcomp I am not able to recall the issue with MS SQL, will this default work there?

File: user-data-store-server/pom.xml [hunk 2]
```diff
@@ -93,6 +93,12 @@
             <artifactId>postgresql</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>com.microsoft.sqlserver</groupId>
+            <artifactId>mssql-jdbc</artifactId>
+            <version>${mssql-jdbc.version}</version>
```
Comment (banterCZ) [inline user-data-store-server/pom.xml:99]: We use the implicit version from Spring Boot parent for postgres. May we use it for MS SLQ as well?

**Review by banterCZ — APPROVED** [review]

File: docs/db/changelog/changesets/user-data-store/1.3.x/20240514-refactor-uds-1.3.0.xml (hunk 1 above)
Comment (zcgandcomp) [reply docs/db/changelog/changesets/user-data-store/1.3.x/20240514-refactor-uds-1.3.0.xml:36]: definition in the change log module
 <property name="now" value="current_timestamp" dbms="mssql"/>

and this construct
 <column name="timestamp_created" type="timestamp" defaultValueDate="${now}" />

should work fine

The type="TIMESTAMP WITHOUT TIME ZONE" check our wiki maybe this works, but I'm not sure.

File: user-data-store-server/pom.xml (hunk 2 above)
Comment (romanstrobl) [reply user-data-store-server/pom.xml]: Ok.

File: docs/db/changelog/changesets/user-data-store/1.3.x/20240514-refactor-uds-1.3.0.xml (hunk 1 above)
Comment (romanstrobl) [reply docs/db/changelog/changesets/user-data-store/1.3.x/20240514-refactor-uds-1.3.0.xml:36]: It should work, it gets generated as e.g.:

`timestamp_created datetime2 CONSTRAINT DF_uds_photo_timestamp_created DEFAULT GETDATE()`

Comment (banterCZ) [reply docs/db/changelog/changesets/user-data-store/1.3.x/20240514-refactor-uds-1.3.0.xml:36]: I found that issue https://github.com/liquibase/liquibase/issues/4644

**Pull request merged**